3주차/FindAnimal/src/App.jsx

```diff
@@ -0,0 +1,445 @@
+import { useState, useEffect } from "react";
+import "./App.css";
+import styled from "styled-components";
+
+function App() {
+  const [showPage0, setShowPage0] = useState(true);
+  const [showPage1, setShowPage1] = useState(false);
+  const [showPage2, setShowPage2] = useState(false);
+  const [showPage3, setShowPage3] = useState(false);
+  const [showPage4, setShowPage4] = useState(false);
+  const [showPageRandom, setShowPageRandom] = useState(false);
+  const [selectedOptions, setSelectedOptions] = useState([]);
+
+  const startGame = (page) => {
+    if (page === "page1") {
+      setShowPage0(false);
+      setShowPage1(true);
+    } else if (page === "pageRandom") {
+      setShowPage0(false);
+      setShowPageRandom(true);
+    }
+  };
+
+  const restartPage1 = () => {
+    setShowPage1(true);
+    setShowPage0(false);
+    setShowPage2(false);
+    setShowPage3(false);
+    setShowPage4(false);
+    setSelectedOptions([]);
+  };
+
+  const goPage0 = (option) => {
+    setShowPage0(true);
+    setShowPage1(false);
+    setShowPage2(false);
+    setShowPage3(false);
+    setShowPage4(false);
+  };
+
+  const goPage1 = (option) => {
+    setShowPage0(false);
+    setShowPage1(true);
+    setShowPage2(false);
+    setShowPage3(false);
+    setShowPage4(false);
+    setSelectedOptions([...selectedOptions, option]);
+  };
+
+  const goPage2 = (option) => {
+    setShowPage0(false);
+    setShowPage1(false);
+    setShowPage2(true);
+    setShowPage4(false);
+    setShowPage3(false);
+    setSelectedOptions([...selectedOptions, option]);
+  };
+
+  const goPage3 = (option) => {
+    setShowPage0(false);
+    setShowPage1(false);
+    setShowPage2(false);
+    setShowPage3(true);
+    setShowPage4(false);
+    setSelectedOptions([...selectedOptions, option]);
+  };
+
+  const goPage4 = (option) => {
+    setShowPage0(false);
+    setShowPage1(false);
+    setShowPage2(false);
+    setShowPage3(false);
+    setShowPage4(true);
+    setSelectedOptions([...selectedOptions, option]);
+  };
```

Comment:
위에서 정안 오빠 말대로 const [showPage, setShowPage]= useState(0); 이런식으로 하나의 상태를 0,1,2로 관리할 수 있다면, 이 함수들도
```suggestion
  const changePage = (page, option) => {
     setShowPage(page);
     if (page === 0) {
     setSelectedOptions([]);
     } else { 
     setSelectedOptions([...selectedOptions, option]);
     }
  };
```
이렇게 간단히 가능하겠다!